crypto/keys/der_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"context"
+	"encoding/base64"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/google/trillian/crypto/keyspb"
+)
+
+func TestPrivateKeyProtoHandler(t *testing.T) {
+	// ECDSA private key in DER format.
+	keyDER, err := base64.StdEncoding.DecodeString("MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgS81mfpvtTmaINn+gtrYXn4XpxxgE655GLSKsA3hhjHmhRANCAASwBWDdgHS04V/cN0LZgc8vZaK4I1HWLLCoaOO27Z0B1aS1aqBE7g1Oo8ldSCBJAvee866kcHhZkVniPdCG2ZZG")
+	if err != nil {
+		t.Fatalf("Could not decode test key: %v", err)
+	}
+
+	sf := NewSignerFactory()
+	sf.AddHandler(&keyspb.PrivateKey{}, NewFromPrivateKeyProto)
+
```

Comment:
Should we test the handler for `*keyspb.Specification`? Ditto for the other public functions.